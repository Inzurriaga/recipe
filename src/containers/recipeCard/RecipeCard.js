import React, { Component } from "react"

export default class RecipeCard extends Component {

  mapIngredients = () => {
    let ingredients = []
    for(var i = 1; i < 21; i++){
      if(this.props.info[`strIngredient${i}`]){
        ingredients.push(
          <div className="ingredient-position">
            <p>{this.props.info[`strIngredient${i}`]}</p>
            <p>{this.props.info[`strMeasure${i}`]}</p>
          </div>
        )
      }
    }
    return ingredients
  }

  render(){
    const ingredients = this.mapIngredients()
    return(
      <article className="recipe-card-container">
        <svg
          className={this.props.svgClass}
          xmlns="http://www.w3.org/2000/svg"
          width="150mm"
          height="50mm"
          viewBox="0 0 150 55">
          <g
            transform="translate(0,-242)">
          <path
            d="m 0,297 c 0,0 0.31935133,-49.16065 30,-50 29.680649,-0.83935 59.287708,-0.83935 90,0 30.71229,0.83935 30,50 30,50 z"/>
          </g>
        </svg>
        <section className="recipe-card-grid">
          <article className="recipe-card-ingredients">
            <div className="labels">
              <h2>Ingredients</h2>
            </div>
            <div className="ingredient-position-labels">
              <h3>Items</h3>
              <h3>Measurements</h3>
            </div>
            {
              ingredients
            }
          </article>
          <article className="recipe-card-image">
          <img src={this.props.info.strMealThumb} alt="hello"/>
          </article>
          <article className="recipe-card-directions">
            <div className="labels">
              <h2>Directions</h2>
            </div>
              <p>{this.props.info.strInstructions}</p>
          </article>
        </section>
      </article>
    )
  }
}