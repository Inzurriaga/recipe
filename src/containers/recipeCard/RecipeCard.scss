.recipe-card-container {
  font-family: 'Roboto', sans-serif;
  position: relative;
  padding: 20px;
  margin-top: 100px;
  z-index: 0;
  width: 80%;
  min-height:100px;
  background-color: #d1d666;
  border-radius: 5px;
  &:nth-child(even) {
    background-color: #b6e1f5;
  }
}

.folder-tap-odd {
  position: absolute;
  top: -80px;
  right: 40px;
  fill: #d1d666;
  width: 250px;
  height: 85px;
}

.folder-tap-even {
  position: absolute;
  top: -80px;
  right: 40px;
  fill: #b6e1f5;
  width: 250px;
  height: 85px;
}

.recipe-card-grid {
  padding: 20px;
  background-color: #ffffff;
  display: grid;
  grid-template-rows: auto auto;
  grid-template-columns: 2fr 1fr;
  grid-template-areas: 
    "ingredients image"
    "directions directions";
}

.recipe-card-ingredients {
  grid-area: ingredients;
}

.recipe-card-image {
  box-shadow: -2px 2px 10px #000000;
  display: flex;
  justify-content: center;
  align-items: center;
  grid-area: image;
  background-color: #c7c7c7;
  width: 410px;
  height: 410px;
  img {
    width: 400px;
    height: 400px;
  }
}

.recipe-card-directions {
  grid-area: directions;
}

.labels {
  padding: 5px;
  position: relative;
  left: -50px;
  clip-path: polygon(90% 0%, 100% 50%, 90% 100%, 0% 100%, 5% 50%, 0% 0%);
  background-color: #de7165;
  width: 250px;
  h2 {
    margin: 0px;
    color: #ffffff;
  }
}

.ingredient-position {
  width: 50%;
  display: flex;
  justify-content: space-between;
}
.ingredient-position-labels {
  text-decoration: underline;
  width: 50%;
  display: flex;
  justify-content: space-between;
}