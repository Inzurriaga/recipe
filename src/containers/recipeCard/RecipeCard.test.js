import React from 'react';
import RecipeCard from './RecipeCard';
import { shallow } from "enzyme";

const mockInfo = {
  strMeal: "pizza",
  strMealThumb: "google.com",
  strInstructions: "add salt"
};

describe("RecipeCard", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <RecipeCard info={mockInfo}/>
    );
  })

  it("should match snap shot", () => {
    expect(wrapper).toMatchSnapshot();
  })
})