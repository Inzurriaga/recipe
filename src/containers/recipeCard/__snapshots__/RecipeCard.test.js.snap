// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match snap shot 1`] = `ShallowWrapper {}`;
